doc/api/stream.md

```diff
@@ -1567,6 +1567,9 @@ added: v10.0.0
 A module method to pipe between streams forwarding errors and properly cleaning
 up and provide a callback when the pipeline is complete.
 
+If the first argument may also be an object or a function returning an object
+implementing the `Symbol.asyncIterator` or `Symbol.iterator`.
```

Comment:
Needs a better description. Might need someone more familiar with the doc format to properly describe this. I've added a description in the PR description on what can be provided but not sure how to best convert that into actual docs. @BridgeAR any suggestions on a good ping?